crates/primitives/src/revm/env.rs

```diff
@@ -109,7 +117,17 @@ pub fn recover_header_signer(header: &Header) -> Option<Address> {
 /// Returns a new [TxEnv] filled with the transaction's data.
 pub fn tx_env_with_recovered(transaction: &TransactionSignedEcRecovered) -> TxEnv {
     let mut tx_env = TxEnv::default();
+
+    #[cfg(not(feature = "optimism"))]
     fill_tx_env(&mut tx_env, transaction.as_ref(), transaction.signer());
+
+    #[cfg(feature = "optimism")]
+    {
+        let mut envelope_buf = Vec::default();
```

Comment:
```suggestion
        let mut envelope_buf = Vec::new();
```

Reply:
Addressed in https://github.com/anton-rs/op-reth/pull/153